lib/plugins/bed.js

```diff
@@ -78,34 +81,39 @@ function inject (bot) {
         }
       }
 
-      if (!bot.canDigBlock(bedBlock)) return cb(new Error('cant click the bed'))
+      if (!bot.canDigBlock(bedBlock)) {
+        return cb(new Error('cant click the bed'))
+      }
 
       const clickRange = [2, -3, -3, 2] // [south, west, north, east]
+      const monsterRange = [7, -8, -8, 7]
       const oppositeCardinal = (metadata.facing + 2) % CARDINAL_DIRECTIONS.length
+
       if (clickRange[oppositeCardinal] < 0) {
         clickRange[oppositeCardinal]--
       } else {
         clickRange[oppositeCardinal]++
       }
 
-      const botPos = bot.entity.position.floored()
-      const nwCorner = headPoint.offset(clickRange[1], -2, clickRange[2]) // North-West lower corner
-      const seCorner = headPoint.offset(clickRange[3], 2, clickRange[0]) // South-East upper corner
-      if (botPos.x > seCorner.x || botPos.x < nwCorner.x || botPos.y > seCorner.y || botPos.y < nwCorner.y || botPos.z > seCorner.z || botPos.z < nwCorner.z) return cb(new Error('the bed is too far'))
-
-      bot.activateBlock(bedBlock, () => {
-        function sleepCb () {
-          clearTimeout(sleepTimeout)
-          cb()
-        }
+      const nwClickCorner = headPoint.offset(clickRange[1], -2, clickRange[2]) // North-West lower corner
+      const seClickCorner = headPoint.offset(clickRange[3], 2, clickRange[0]) // South-East upper corner
+      if (botPos.x > seClickCorner.x || botPos.x < nwClickCorner.x || botPos.y > seClickCorner.y || botPos.y < nwClickCorner.y || botPos.z > seClickCorner.z || botPos.z < nwClickCorner.z) {
```

Comment:
Are you sure it's a square around and not radius?

Reply:
I've been doing some testing, so I'm quite sure